addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2628,6 +2631,79 @@ export class OdooEditor extends EventTarget {
     // TABLE MANAGEMENT
     // ================
 
+    /**
+     * Sets selection in table cells in order to select the cells
+     * rectangularly when pressing shift + arrow key (just like
+     * google docs).
+     *
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    _setSelectionInTable(ev) {
+        const selection = this.document.getSelection();
+        const startTable = closestElement(selection.anchorNode, 'table');
+        const endTable = closestElement(selection.focusNode, 'table');
+        if (!(startTable && endTable) || startTable !== endTable) {
+            return;
+        }
+        handleMultipleRanges(this.editable, { select: true, correctTripleClick: selection.anchorNode && selection.anchorNode.nodeName === 'TR' });
+        const [startTd, endTd] = [closestElement(selection.anchorNode, 'td'), closestElement(selection.focusNode, 'td')];
+        const endCellPosition = { x: getRowIndex(endTd), y: getColumnIndex(endTd) };
+        const tds = [...startTable.querySelectorAll('tr')]
+                    .filter(tr => closestElement(tr, 'table') === startTable)
+                    .map(tr => [...tr.querySelectorAll('td')].filter(td => closestElement(td, 'table') === startTable));
+        let targetTd, targetNode;
+        switch (ev.key) {
+            case 'ArrowUp': {
+                if (endCellPosition.x > 0) {
+                    targetTd = tds[endCellPosition.x - 1][endCellPosition.y];
+                } else {
+                    if (previousLeaf(startTable, this.editable)) {
+                        targetNode = previousLeaf(startTable, this.editable);
+                    } else {
+                        ev.preventDefault();
+                        return;
+                    }
+                }
+                break;
+            }
+            case 'ArrowDown': {
+                if (endCellPosition.x < tds.length - 1) {
+                    targetTd = tds[endCellPosition.x + 1][endCellPosition.y];
+                } else {
+                    targetNode = nextLeaf(startTable, this.editable);
+                }
+                break;
+            }
+            case 'ArrowRight': {
+                if (endCellPosition.y === tds[0].length - 1) {
+                    ev.preventDefault();
+                    return;
+                } else {
+                    targetTd = tds[endCellPosition.x][endCellPosition.y + 1];
```

Comment:
Let's ensure consistency by setting targetTd within the if, for both `ArrowRight` and `ArrowLeft` just like you did it for `ArrowUp` and `ArrowDown`.